src/opm/input/eclipse/Parser/ParserKeyword.cpp

```diff
@@ -845,26 +871,65 @@ void set_dimensions( ParserItem& item,
         }
     }
 
+    void ParserKeyword::setMatchRegexSuffix(const std::string& deckNameRegexp) {
+        try {
+            const auto sfx = std::regex { deckNameRegexp };
+            this->m_matchRegexSuffix = deckNameRegexp;
+        }
+        catch (const std::exception& e) {
+            std::cerr << fmt::format("'{}' is not a valid regular expression suffix for keyword '{}':\n"
+                                     "\t{}\nSuffix is ignored.", deckNameRegexp, this->getName())
+                      << '\n';
+        }
+    }
+
     bool ParserKeyword::matches(const std::string_view& name ) const {
-        if (!validDeckName(name ))
+        if (!validDeckName(name))
             return false;
 
-        else if( m_deckNames.count( std::string(name) ) )
+        else if (matchesDeckNames(name))
+            return true;
+
+        else if (hasMatchRegex() && std::regex_match(std::string{name}, m_matchRegex))
+            return true;
+
+        return false;
+    }
+
+    bool ParserKeyword::matchesDeckNames(std::string_view name) const {
+        const auto nameStr = std::string { name };
+
+        if (m_deckNames.find(nameStr) != m_deckNames.end()) {
             return true;
+        }
 
-        else if (hasMatchRegex()) {
-            bool match = std::regex_match( name.begin(), name.end(), m_matchRegex);
-            if (match)
-                return true;
+        auto matchRegex = [&nameStr, this](const auto& makeNameREString)
```

Comment:
I found the passing of two different lambdas to this lambda a bit confusing, if you test for both conditions in the std::any_of() call you do not even need to make the lambdas. I suggest:
```c++
 return std::any_of(this->m_deckNames.begin(),
                    this->m_deckNames.end(),
                    [&nameStr](const std::string& deckName)
                    {
                        return std::regex_match(nameStr, std::regex{ deckName })
                            || std::regex_match(nameStr, std::regex{ deckName + m_matchRegexSuffix });
                    });
```
Assuming that I understood the purpose of the code!

Reply:
I wrote it the original way because I wanted to eventually remove the "deck name as regular expression" logic.  I still think that would be a good idea, but I'm not going to push for it in the short term.  I've rewritten the code to use a single call to `std::any_of()` as you've outlined.